### PR TITLE
[TECH-676] Use properly LatestConfig to encrypt affidavit

### PR DESCRIPTION
**Pull request by w0st**

### CODE REVIEW

File: lib/av_client/types.ts [hunk 1]
```diff
@@ -272,7 +272,6 @@ export interface LatestConfig {
   items: LatestConfigItems
   receipt?: string
   affidavit?: AffidavitConfig
-  castRequestItemAttachmentEncryptionKey?: string
```
Comment (w0st) [inline lib/av_client/types.ts:275]: This element is not available on the root object.

**Review by unnunhexium — APPROVED** [review]

**Review by av-emil — APPROVED** [review]

File: lib/av_client.ts [hunk 2]
```diff
@@ -396,9 +396,9 @@ export class AVClient implements IAVClient {
 
       let encryptedAffidavit;
 
-      if (affidavit && this.latestConfig && this.latestConfig.castRequestItemAttachmentEncryptionKey) {
+      if (affidavit && this?.latestConfig?.items?.electionConfig?.content?.castRequestItemAttachmentEncryptionKey) {
```
Comment (av-lukas) [inline lib/av_client.ts:399]: Given we're using TypeScript, can we make these `null` assertions at the Type level? (i.e., fail when parsing the API response)

Comment (w0st) [reply lib/av_client.ts:399]: @MateuszMichalowski @unnunhexium WDYT?

Comment (MateuszMichalowski) [reply lib/av_client.ts:399]: Are you referring to add union types with null possibility to these @av-lukas or you have something else in mind?

Comment (av-lukas) [reply lib/av_client.ts:399]: I'm not sure how union types would be used here, but I'm referring to the fact that at this point in the lifecycle of the av_client, if the affidavit is present, then it also means that the client has been told to ask for one, via the configuration object. So, in that case, the `castRequestItemAttachmentEncryptionKey` should always be present, and the null checks shouldn't be necessary. The class that wraps it should enforce it to be present when parsing the request.

**Pull request merged**